api/src/main/java/org/openmrs/module/metadatamapping/api/MetadataMappingService.java

```diff
@@ -419,4 +423,167 @@
+	 * @param retiredHandlingMode handle retired objects using this mode
+	 * @return list of members in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @since 1.2
+	 * @should get members in desired order 1
+	 * @should respect retire fetch mode 1
+	 */
+	List<MetadataSetMember> getMetadataSetMembers(MetadataSet metadataSet, int firstResult, int maxResults,
+	        RetiredHandlingMode retiredHandlingMode);
+	
+	/**
+	 * Get members of a metadata set with given uuid. If members have {@link MetadataSetMember#getSortWeight()} set they will be ordered
+	 * in ascending order according to said weight. Note that due to differences in database implementations, the order
+	 * will be unpredictable, if there are null sort weights in the set.
+	 * @param metadataSetUuid metadata set uuid
+	 * @param firstResult zero based index of first result to get
+	 * @param maxResults maximum number of results to get
+	 * @param retiredHandlingMode handle retired objects using this mode
+	 * @return list of members in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @since 1.2
+	 * @should get members in desired order 1
+	 * @should respect retire fetch mode 1
+	 */
+	List<MetadataSetMember> getMetadataSetMembers(String metadataSetUuid, int firstResult, int maxResults,
+	        RetiredHandlingMode retiredHandlingMode);
+	
+	/**
+	 * Get unretired metadata items in the set of specified type. If set members have {@link MetadataSetMember#getSortWeight()} set they will
+	 * be ordered in ascending order according to said weight. Note that due to differences in database implementations,
+	 * the order  will be unpredictable, if there are null sort weights in the set.
+	 * @param type type of the metadata items
+	 * @param metadataSet metadata set
+	 * @param firstResult zero based index of first result to get
+	 * @param maxResults maximum number of results to get
+	 * @param <T> type of the metadata items
+	 * @return list of items in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @since 1.2
+	 * @should get unretired metadata items of unretired terms matching type in sort weight order 1
+	 * @should throw IllegalArgumentException if set does not exist
+	 */
+	<T extends OpenmrsMetadata> List<T> getMetadataSetItems(Class<T> type, MetadataSet metadataSet, int firstResult,
+	        int maxResults);
+	
+	/**
+	 * Get metadata item referred to by the given metadata term mapping
+	 * @param type type of the metadata item
+	 * @param setMember metadata set member referring to object
+	 * @param <T> type of the metadata item
+	 * @return metadata item or null, if not found or if either the set member or the metadata item itself are
+	 * retired
+	 * @should return unretired metadata item for unretired set member
+	 * @should not return retired metadata item for unretired set member
+	 * @should not return unretired metadata item for retired set member
```

Comment:
It should return unretired and retired metadata regardless whether set member is retired or not.